servicetalk-capacity-limiter-api/src/main/java/io/servicetalk/capacity/limiter/api/GradientCapacityLimiter.java

```diff
@@ -125,23 +125,29 @@ public String name() {
 
     @Override
     public Ticket tryAcquire(final Classification classification, @Nullable final ContextMap meta) {
-        int newPending;
-        int newLimit;
-
         Ticket ticket = null;
         lock.lock();
         try {
-            newLimit = (int) limit;
+            final int newLimit = (int) limit;
             if (pending < limit) {
-                newPending = ++pending;
+                final int newPending = pending + 1;
                 ticket = new DefaultTicket(this, newLimit - newPending, newPending);
+                // update the state last just in case anything throws.
+                ++pending;
             }
+        } catch (Throwable t) {
+            LOGGER.error("Exception while attempting to acquire ticket", t);
         } finally {
             lock.unlock();
         }
 
         if (ticket != null) {
-            observer.onActiveRequestsIncr();
+            try {
+                observer.onActiveRequestsIncr();
+            } catch (Throwable t) {
```

Comment:
Everywhere in ServiceTalk we guarantee that observers never alter the execution. Here we honor this by always wrapping a user-provided `Observer` with `GradientCapacityLimiter.CatchAllObserver`.
This helps us to avoid try-catching in every place around the codebase.